lib/types.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+
+export type Range = {
+    start: number,
+    end: number,
+}
+
+
+export type Resolved = {
+    url?: string,
+    filename?: string,
+}
+
+export type Destination = {|
+    start: number,
+    end: number,
+|}
+
+type ExternalModule = {
+    local: string,
+    start: number,
+    end: number,
+    moduleName: string,
+    imported: string,
+}
+
+type SuggestionFromImport = {|
+    type: 'from-import',
+    moduleName: string,
+    imported: string,
+    bindingStart: number,
+    bindingEnd: number,
+|}
+
+type BindingSuggestion = {|
+    type: 'binding',
+    start: number,
+    end: number,
+    moduleName: typeof undefined,
```

Comment:
Without it I was having trouble with `resolveModule`. My next commit will change `resolveModule` to check the `.type` and throw instead of checking for the presence of `.moduleName`